src/tfvc/tfvc-extension.ts

```diff
@@ -48,25 +49,57 @@ export class TfvcExtension  {
             const changeset: string =
                 await this._repo.Checkin(checkinInfo.files, checkinInfo.comment, checkinInfo.workItemIds);
             TfvcOutput.AppendLine("Changeset " + changeset + " checked in.");
+            TfvcSCMProvider.ClearCheckinMessage();
         } catch (err) {
             this._manager.DisplayErrorMessage(err.message);
         }
     }
 
-    public async TfvcExclude(): Promise<void> {
-        //
-    }
+    public async TfvcExclude(uri?: Uri): Promise<void> {
+        if (!this._manager.EnsureInitialized(RepositoryType.TFVC)) {
+            this._manager.DisplayErrorMessage();
+            return;
+        }
 
-    public async TfvcExcludeAll(): Promise<void> {
-        //
+        if (uri) {
+            //Keep an in-memory list of items that were explicitly excluded. The list is not persisted at this time.
+            await TfvcSCMProvider.Exclude(TfvcSCMProvider.GetPathFromUri(uri));
+        }
     }
 
-    public async TfvcInclude(): Promise<void> {
-        //
+    public async TfvcInclude(uri?: Uri): Promise<void> {
+        if (!this._manager.EnsureInitialized(RepositoryType.TFVC)) {
+            this._manager.DisplayErrorMessage();
+            return;
+        }
+
+        if (uri) {
+            let resource: Resource = TfvcSCMProvider.ResolveTfvcResource(uri);
+            let path: string = TfvcSCMProvider.GetPathFromUri(uri);
+
+            //At this point, an unversioned file could be a candidate file, so call Add.  Once it is added, it should be a Pending change.
+            if (!resource.IsVersioned) {
+                await this._repo.Add([path]);
+                //Even if adding a file, Unexclude it (it may have been excluded previously)
```

Comment:
The comments are a bit confusing. Looks like you used to have an else clause here and then removed it.